src/app/pages/user/dialogs/base/user-details-form/user-details-form.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/observable/of';
+import { classTestData } from './../../../../../../mocks/assets/classes.mock';
+import { UserService } from './../../../services/user.service';
+import { Apollo } from 'apollo-angular';
+import { ClassService } from './../../../../class/services/class.graphql.service';
+import { FormsModule, ReactiveFormsModule, FormBuilder } from '@angular/forms';
+import { NO_ERRORS_SCHEMA } from '@angular/compiler/src/core';
+import { UserDetailsFormComponent } from './user-details-form';
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+import { User, UserType } from './../../../../../models/user.model';
+import { Class } from '../../../../../models/class.model';
+
+describe('student component', () => {
+  let dataMock: User;
+  let fixture: ComponentFixture<UserDetailsFormComponent>;
+  let component: UserDetailsFormComponent;
+  let classServiceMock: Partial<ClassService>;
+
+  beforeEach(async () => {
+    classServiceMock = {
+      getAllClasses: jest.fn().mockReturnValue(Observable.of(classTestData.classes)),
+      delete: jest.fn(),
+      create: jest.fn(),
+      update: jest.fn(),
+    };
+
+    dataMock = {
+      email: 'a',
+      firstname: 'a',
+      lastname: 'a',
+      password: 'a',
+      role: UserType.PRINCIPLE,
+      username: 'a',
+      _id: 0,
+      class: new Class(),
+    };
+
+    TestBed.configureTestingModule({
+      imports: [
+        ReactiveFormsModule,
+        // MatFormFieldModule,
```

Comment:
same here, can you please remove commented code?

Reply:
Done